addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,463 @@
+
+    private void updateNextTimer(MowerInfo info) {
+        State dateTime = convertUnixToDateTimeType(info.getTimer().getNext().getUnix());
+        updateState(CHANNEL_TIMER_NEXT_TIMER, dateTime);
+    }
+    
+
+    private State convertUnixToDateTimeType(String unixTimeSec) {
+        Instant ns = Instant.ofEpochMilli(Long.valueOf(unixTimeSec) * 1000);
+        ZonedDateTime zdt = ZonedDateTime.ofInstant(ns, ZoneId.of("UTC"));
+        return new DateTimeType(zdt);
+    }
+
+    private void refreshVersionInfo() {
+        VersionInfo info = robonectClient.getVersionInfo();
+        if (info.isSuccessful()) {
+            Map<String, String> properties = editProperties();
+            properties.put(Thing.PROPERTY_SERIAL_NUMBER, info.getRobonect().getSerial());
+            properties.put(Thing.PROPERTY_FIRMWARE_VERSION, info.getRobonect().getVersion());
+            properties.put(PROPERTY_COMPILED, info.getRobonect().getCompiled());
+            properties.put(PROPERTY_COMMENT, info.getRobonect().getComment());
+            updateProperties(properties);
+        } else {
+            logger.debug("Could not retrieve mower version info. Robonect error response message: {}",
+                         info.getErrorMessage());
+        }
+    }
+
+    private void refreshLastErrorInfo() {
+        ErrorList errorList = robonectClient.errorList();
+        if (errorList.isSuccessful()) {
+            if (errorList.getErrors() != null && errorList.getErrors().size() > 0) {
+                ErrorEntry lastErrorEntry = errorList.getErrors().get(0);
+                updateLastErrorChannels(lastErrorEntry);
+            }
+        } else {
+            logger.debug("Could not retrieve mower error list. Robonect error response message: {}",
+                         errorList.getErrorMessage());
+        }
+    }
+
+    private void updateLastErrorChannels(ErrorEntry error) {
+        if (error.getErrorMessage() != null) {
+            updateState(CHANNEL_LAST_ERROR_MESSAGE, new StringType(error.getErrorMessage()));
+        }
+        if (error.getErrorCode() != null) {
+            updateState(CHANNEL_LAST_ERROR_CODE, new DecimalType(error.getErrorCode().intValue()));
+        }
+        if (error.getDate() != null) {
+            State dateTime = convertUnixToDateTimeType(error.getUnix());
+            updateState(CHANNEL_LAST_ERROR_DATE, dateTime);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        RobonectConfig robonectConfig = getConfigAs(RobonectConfig.class);
+        RobonectEndpoint endpoint = new RobonectEndpoint(robonectConfig.getHost(), robonectConfig.getUser(),
+                                                         robonectConfig.getPassword());
+        httpClient = new HttpClient();
+        try {
+            httpClient.start();
+            robonectClient = new RobonectClient(httpClient, endpoint);
+        } catch (Exception e) {
+            logger.error("Exception when trying to initialize", e);
```

Comment:
it's the httpClient.start() which may throw an exception here... I guess with the shared httpClient instance this would disappear as well, right. Just still the question if PreEmptive Authentication counts as "specific configuration requirements" and therefore I cannot use it. In this case I understand I would throw the exception. Can I wrap it into a RuntimeException?

Reply:
Yeah, wrapping it in a RuntimeException should be fine in this case (as the Exception from Jetty is anyhow very unspecific).

Reply:
I don't know if it's related here, but every time the binding tries to poll my mower and doesn't succeed I get an error in the log, "Fatal transport error: java.util.concurrent.TimeoutException". Not being able to reach it should be considered quite normal, classing it as an error doesn't seem right.

Reply:
Sounds like a good example why more finegrained exceptions should be caught and handled differently - a TimeoutException should indeed set the Thing to OFFLINE-COMMUNICATION_ERROR and do a periodic retry.

Reply:
Not too sure about that @kaikreuzer . There is a setting for offline timeout at Thing level, described by @reyem as "The maximum time the mower may be offline before the offline trigger is triggered." The mower being temporarily outside of range is normal and it's not an error unless it's not still unreachable after this timeout.

Reply:
actually this will not happen anymore. This was just possible before, as I was calling refreshInfo already the first time here in a previous version in the binding. But the refactoring removed this. So that should be ok.. now